tests/SprykerSdkTest/AsyncApi/Console/SchemaMessageAddConsoleTest.php

```diff
@@ -100,4 +101,67 @@ public function testAddMessageReturnsErrorCodeAndPrintsErrorMessagesWhenMessageT
             ['verbosity' => OutputInterface::VERBOSITY_VERBOSE],
         );
     }
+
+    /**
+     * This test ensures that a message can only be added once per channel.
+     * The given API file already has the message that should be added and thus adding it again will be skipped.
+     *
+     * @return void
+     */
+    public function testAddMessageOnlyWhenMessageNameDoesNotExistInChannel(): void
+    {
+        // Arrange
+        $commandTester = $this->tester->getConsoleTester(SchemaMessageAddConsole::class, false);
+
+        // Act
+        // The message 'OutgoingMessage' is added for the second time in the channel 'channelNameA'
+        $commandTester->execute(
+            [
+                '--' . SchemaMessageAddConsole::OPTION_MESSAGE_TYPE => 'subscribe',
+                '--' . SchemaMessageAddConsole::OPTION_PROPERTY => ['property:string'],
+                '--' . SchemaMessageAddConsole::OPTION_ASYNC_API_FILE => codecept_data_dir('api/asyncapi/asyncapi.yml'),
+                SchemaMessageAddConsole::ARGUMENT_CHANNEL_NAME => 'channelNameA',
+                SchemaMessageAddConsole::ARGUMENT_OPERATION_ID => 'operationId',
+                SchemaMessageAddConsole::ARGUMENT_MESSAGE_NAME => 'OutgoingMessage',
+            ],
+        );
+
+        // The YAML file is parsed after the message was added
+        $asyncApi = Yaml::parseFile(codecept_data_dir('api/asyncapi/asyncapi.yml'));
+
+        // Assert
+        // The message exists only once in the channel
+        $this->tester->assertMessageExistsOnlyOnceInChannel($asyncApi, 'OutgoingMessage', 'channelNameA', 'subscribe');
```

Comment:
The assert method name is good and explains enough no need for this inline comment.